sbc_compassion/models/import_letter_line.py

```diff
@@ -53,36 +55,55 @@ class ImportLetterLine(models.Model):
     @api.depends('partner_codega', 'child_code', 'sponsorship_id',
                  'supporter_languages_id')
     def _check_status(self):
-        """
-        At each change, check if all the fields are OK
+        """ At each change, check if all the fields are OK
         """
         default_template = self.env.ref('sbc_compassion.default_template')
         for line in self:
-            if line.sponsorship_status or line.is_encourager:
-                line.status = "encourager"
-            if line.sponsorship_status is True:
-                line.status = "sponsor"
+            if not line.sponsorship_found:
+                line.status = "no_sponsorship"
+                if not line.child_partner_found:
+                    line.status = "no_child_partner"
             elif not line.template_id or (line.template_id.id ==
                                           default_template.id):
-                line.status = "temp"
+                line.status = "no_template"
             elif len(line.supporter_languages_id) != 1:
-                line.status = "lang"
+                line.status = "no_lang"
             else:
                 line.status = "ok"
 
     @api.multi
     @api.depends('partner_codega', 'child_code')
     def _set_sponsorship_id(self):
+        """ From the partner codega and the child code, find the record
+        linking them together.
+        At the same time, check if the child, the partner and the sponsorship
+        are found.
+        """
         for line in self:
             if line.partner_codega and line.child_code:
                 line.sponsorship_id = line.env['recurring.contract'].search([
                     ('child_id.code', '=', line.child_code),
                     ('partner_codega', '=', line.partner_codega),
                     ('is_active', '=', True)], order='end_date desc', limit=1)
-                if len(line.sponsorship_id) == 1:
-                    line.sponsorship_status = None
+                if not line.sponsorship_id:
+                    line.sponsorship_id = line.env[
+                        'recurring.contract'].search(
+                            [('child_id.code', '=', line.child_code),
+                             ('partner_codega', '=', line.partner_codega)],
+                            order='start_date desc', limit=1)
```

Comment:
I tried without, but if someone is dropping a sponsorship and want to send a last letter, it won't be possible without the second one